ibis/expr/rules.py

```diff
@@ -424,11 +424,17 @@ def table(arg, *, schema=None, **kwargs):
 
 @validator
 def column_from(name, column, *, this):
-    if not isinstance(column, (str, int)):
-        raise com.IbisTypeError(
-            f"value must be an int or str, got {type(column).__name__}"
-        )
-    return getattr(this, name)[column]
+    if isinstance(column, ir.AnyColumn) and hasattr(column.op(), "name"):
+        maybe_column = column.op().name
+        if column.equals(getattr(this, name)[maybe_column]):
+            return column
+    elif isinstance(column, (str, int)):
+        return getattr(this, name)[column]
```

Comment:
Can you pull up `getattr(this, name)` into a variable (it's a table)? No reason to repeat it.